crates/ibc/src/core/ics04_channel/handler/timeout.rs

```diff
@@ -105,65 +77,100 @@ pub fn process<Ctx: ChannelReader>(
         });
     }
 
-    let result = if source_channel_end.order_matches(&Order::Ordered) {
-        if packet.sequence < msg.next_sequence_recv {
-            return Err(PacketError::InvalidPacketSequence {
-                given_sequence: packet.sequence,
-                next_sequence: msg.next_sequence_recv,
+    // Verify proofs
+    {
+        let client_id_on_a = conn_end_on_a.client_id().clone();
+        let client_state_of_b_on_a = ctx_a
+            .client_state(&client_id_on_a)
+            .map_err(PacketError::Channel)?;
+
+        // check that timeout height or timeout timestamp has passed on the other end
+        let proof_height = msg.proof_height_on_b;
+
+        if packet.timeout_height.has_expired(proof_height) {
+            return Err(PacketError::PacketTimeoutHeightNotReached {
+                timeout_height: packet.timeout_height,
+                chain_height: proof_height,
             });
         }
-        verify_next_sequence_recv(
-            ctx,
-            msg.proofs.height(),
-            &connection_end,
-            packet.clone(),
-            msg.next_sequence_recv,
-            &msg.proofs,
-        )
-        .map_err(PacketError::Channel)?;
 
-        source_channel_end.state = State::Closed;
-        PacketResult::Timeout(TimeoutPacketResult {
-            port_id: packet.source_port.clone(),
-            channel_id: packet.source_channel.clone(),
-            seq: packet.sequence,
-            channel: Some(source_channel_end.clone()),
-        })
-    } else {
-        verify_packet_receipt_absence(
-            ctx,
-            msg.proofs.height(),
-            &connection_end,
-            packet.clone(),
-            &msg.proofs,
-        )
-        .map_err(PacketError::Channel)?;
+        let consensus_state_of_b_on_a = ctx_a
+            .client_consensus_state(&client_id_on_a, &proof_height)
+            .map_err(PacketError::Channel)?;
+        let proof_timestamp = consensus_state_of_b_on_a.timestamp();
 
-        PacketResult::Timeout(TimeoutPacketResult {
-            port_id: packet.source_port.clone(),
-            channel_id: packet.source_channel.clone(),
-            seq: packet.sequence,
-            channel: None,
-        })
-    };
+        let packet_timestamp = packet.timeout_timestamp;
```

Comment:
Can we remove this variable?

Reply:
[b9eb033](https://github.com/cosmos/ibc-rs/pull/323/commits/b9eb0335bce88634fb00609d1088ce620791aab6)